neon_llm_core/chatbot.py

```diff
@@ -73,84 +78,115 @@ def ask_discusser(self, options: dict, context: dict = None) -> str:
         """
+        # Determine the relevant prompt
+        prompt_id = context.get('prompt_id') if context else None
+        prompt_sentence = None
+        if prompt_id:
+            prompt_sentence = self.prompt_id_to_shout.get(prompt_id)
+
+        # Remove self answer from available options
+        options = {k: v for k, v in options.items()
+                   if k != self.service_name}
+
         if options:
-            options = {k: v for k, v in options.items()
-                       if k != self.service_name}
             bots = list(options)
             bot_responses = list(options.values())
             LOG.info(f'bots={bots}, answers={bot_responses}')
-            prompt = self.prompt_id_to_shout.pop(context['prompt_id'], '')
-            answer_data = self._get_llm_api_choice(prompt=prompt,
+            answer_data = self._get_llm_api_choice(prompt=prompt_sentence,
                                                    responses=bot_responses)
             LOG.info(f'Received answer_data={answer_data}')
-            sorted_answer_indexes = answer_data.get('sorted_answer_indexes')
-            if sorted_answer_indexes:
-                return bots[sorted_answer_indexes[0]]
+            if answer_data and answer_data.sorted_answer_indexes:
+                return bots[answer_data.sorted_answer_indexes[0]]
         return "abstain"
 
-    def _get_llm_api_response(self, shout: str) -> dict:
+    def _get_llm_api_response(self, shout: str) -> Optional[LLMProposeResponse]:
         """
         Requests LLM API for response on provided shout
-        :param shout: provided should string
-        :returns response string from LLM API
+        :param shout: Input prompt to respond to
+        :returns response from LLM API
         """
         queue = self.mq_queue_config.ask_response_queue
         LOG.info(f"Sending to {self.mq_queue_config.vhost}/{queue}")
         try:
-            return send_mq_request(vhost=self.mq_queue_config.vhost,
-                                   request_data={"query": shout,
-                                                 "history": [],
-                                                 "persona": self.persona},
-                                   target_queue=queue,
-                                   response_queue=f"{queue}.response")
+            request_data = LLMProposeRequest(model=self.base_llm,
+                                             persona=self.persona,
+                                             query=shout,
+                                             history=[],
+                                             message_id="")
+            resp_data = send_mq_request(vhost=self.mq_queue_config.vhost,
+                                        request_data=request_data.model_dump(),
+                                        target_queue=queue,
+                                        response_queue=f"{queue}.response")
+            return LLMProposeResponse(**resp_data)
```

Comment:
I would omit kwargs-like argument passing here, to make it more flexible to any potential updates to the callback body of `send_mq_request`